### ISSUE TITLE
cannot pass custom transport_params to s3_iter_bucket  api

### ISSUE THREAD

**Issue by indrajeet307** [piskvorky/smart_open#346]
I cannot pass custom transport_params to s3_iter_bucket  api, It defaults to using amazon s3. 
I am trying smart_open with a locally deployed minio server, that supports s3 like access.

**Comment by mpenkov** [comment]
Yes, this is currently not supported. Are you able to make a PR?

**Comment by indrajeet307** [comment]
The changes, I made, in s3.py file
1. Add optional resource_kwargs to iter_bucket api
2. _list_bucket also accepts optional resource_kwargs, and creates a client with or without it.

I tried writing test but moto did not allow sending endpoint_url.
I am getting authentication failed while trying to push commit.

**Comment by mpenkov** [comment]
You won't be able to push to our repository. That's by design. You need to fork our repository, push the changes to your own fork, and then open a pull request. Then we'll review the PR and merge your changes in.

Here's a guide that describes what you need to do: https://guides.github.com/activities/forking/

Let me know if you're up for it!

**Comment by mpenkov** [comment]
Duplicates #259

**Comment by maranqz** [comment]
It's not duplication because [calling client](https://github.com/RaRe-Technologies/smart_open/blob/1c5c56cc7081a90df1b1fe64cd6314999434ee1e/smart_open/s3.py#L1017) doesn't sending endpoint_url to session.client('s3', ....).

**Comment by ddelange** [comment]
> Duplicates [#259](https://github.com/piskvorky/smart_open/issues/259)

#372
